<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Genetic Relatedness Analysis</title>
    <style>
        /* Add your CSS styles here */
        body {
            font-family: Arial, sans-serif;
            background-color: #f8f8f8;
            color: #333;
            margin: 0;
            padding: 0;
        }
        
        h1 {
            text-align: center;
            color: #006699;
        }
        
        #tabs {
            text-align: center;
            margin: 20px 0;
        }
        
        .tab {
            display: none;
            padding: 20px;
            background-color: #fff;
            border-radius: 8px;
            box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
        }
        
        button {
            background-color: #006699;
            color: #fff;
            border: none;
            padding: 10px 20px;
            border-radius: 5px;
            cursor: pointer;
            transition: background-color 0.3s;
        }
        
        button:hover {
            background-color: #005580;
        }
        
        .active {
            background-color: #005580;
        }
        
        .active:hover {
            background-color: #004466;
        }
        
        .explanation {
            color: #006699;
            margin-bottom: 20px;
        }
        
        .loading-bar {
            width: 100%;
            background-color: #f3f3f3;
            margin-top: 10px;
            border-radius: 5px;
            display: none;
        }
        
        .progress-bar {
            width: 0;
            height: 20px;
            background-color: #4caf50;
            border-radius: 5px;
            transition: width 2s linear;
        }
        
        .command-block {
            background-color: #f3f3f3;
            padding: 10px;
            border-radius: 5px;
            margin-bottom: 20px;
            font-family: monospace;
        }
    </style>
</head>
<body>
    <h1>Genetic Relatedness Analysis</h1>
    
    <!-- Tab navigation -->
    <div id="tabs">
<button onclick="openTab('background')" id="btnBackground">Background</button>
<button onclick="openTab('methods')" id="btnMethods">Methods & Analysis</button>
<button onclick="openTab('plink')" id="btnPlink">PLINK Output Explained</button>


    </div>

<!-- Background Section -->
<div id="background" class="tab">
    <h2>Background</h2>
    <div class="explanation">
        <p><strong>What is DNA?</strong></p>
        <p>DNA, or deoxyribonucleic acid, is a molecule that contains the genetic instructions used in the development and functioning of all known living organisms. Single Nucleotide Polymorphisms (SNPs) are variations at a single position in a DNA sequence among individuals. These variations can be used to predict certain genetic traits, predispositions to diseases, and ancestral backgrounds.</p>
        <!-- Add an image to illustrate DNA -->
        <img src="https://medlineplus.gov/images/PX000098_PRESENTATION.jpeg" alt="DNA" style="width: 300px; height: auto;">

        <p><strong>Every Human is Unique</strong></p>
        <p>Every generation we make a few changes in our DNA that gets passed on to our offspring.  For about 12,000 generations humans have been making these changes and today the average difference between two random people is about 0.1 percent.  That is a small number until you consider that our genomes are 3 billion nucleotides (G, A, T, C) long.  So that’s an average of 3 million differences.  These differences, sometimes called mutations, generally do not cause any illness or noticeable change in function, but they remain as a marker of the person they happened in and all their descendance that will now carry that mutation.  These differences are critical to defining regions of chromosomes that are shared because of inheritance from one generation to another a concept called "Identity by descent" or (IBD).  These changes can also point to where (what historic population) the mutation arose.  </p>

        <p><strong>Understanding SNPs and Genetic Variation</strong></p>
        <p>Scientists have determined the complete DNA sequences of people from all over the world with a goal of understanding all the similarities and differences that have accumulated among people.  Many of these differences are called SNPs for Single Nucleotide Polymorphisms.  These are positions on your chromosomes where some people have one nucleotide (like G) and others have a different nucleotide like (T).  Based on that information they have identified many positions on every chromosome that are different in different people and can be identified in a simple accurate test that starts with some cheek cells.  That test tells you what nucleotides are at ~1 million positions in your (two) genomes.</p>
        
        <p><strong>You are really two humans</strong></p>
        <p>½ of your DNA is from your mother and ½ from your father. If you look at your genome you will be 50% identical to each of your parents. Each of your parents share ½ of their DNA with each of your grandparents.  So you share 25% of your DNA with each of your 4 grandparents.</p>
        <img src="https://gcbias.org/wp-content/uploads/2013/11/family_tree_w_trans.png" alt="Haplotypes" style="width: 600px; height: auto;">


        <p><strong>What are Haplotypes?</strong></p>
        <p>A haplotype is a group of genes within an organism that was inherited together from a single parent. The term also refers to a combination of alleles or a set of SNPs found on the same chromosome that tend to be inherited together. Haplotypes can be used to identify genetic ancestry and are crucial in studies of human evolutionary history.</p>
        <!-- Add an image to illustrate Haplotypes -->
        <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/2/2e/Dna-SNP.svg/363px-Dna-SNP.svg.png" alt="Haplotypes" style="width: 300px; height: auto;">

        <p><strong>Data Collection for Genetic Relatedness analysis</strong></p>
        <p>Genetic relatedness data is collected through DNA sequencing or genotyping. Some companies send DNA testing kits where users send a DNA sample, typically saliva, for analysis. The data returned includes information on SNPs and haplotypes, which serve as genetic markers.</p>

        <p><strong>Data File Format</strong></p>
        <p>The DNA sequenicng data is typically in a file format containing positions on the genome inherited from both parents. This data allows for the analysis of genetic relatedness and ancestry composition. It usually includes a vast amount of SNPs which can be analyzed to provide insights into a person's ancestry and potential health predispositions.</p>
    
<table border="1" cellspacing="0" cellpadding="5">
    <thead>
        <tr>
            <th>SNP ID</th>
            <th>Allele 1</th>
            <th>Allele 2</th>
            <th>Chromosome</th>
            <th>Genomic Position</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>rs7419119</td>
            <td>A</td>
            <td>G</td>
            <td>12</td>
            <td>842013</td>
        </tr>
        <tr>
            <td>rs1234567</td>
            <td>C</td>
            <td>T</td>
            <td>7</td>
            <td>1205543</td>
        </tr>
        <tr>
            <td>rs9876543</td>
            <td>G</td>
            <td>A</td>
            <td>3</td>
            <td>4421889</td>
        </tr>
        <tr>
            <td>rs5678901</td>
            <td>T</td>
            <td>C</td>
            <td>1</td>
            <td>100101</td>
        </tr>
        <tr>
            <td>rs2468013</td>
            <td>A</td>
            <td>G</td>
            <td>5</td>
            <td>5050505</td>
        </tr>
        <tr>
            <td>rs1357924</td>
            <td>G</td>
            <td>A</td>
            <td>16</td>
            <td>1609999</td>
        </tr>
        <tr>
            <td>rs1122334</td>
            <td>C</td>
            <td>T</td>
            <td>22</td>
            <td>2202202</td>
        </tr>
    </tbody>
</table>


</div>




</div>


    <!-- Methods & Analysis Section -->
    <div id="methods" class="tab">
        <h2>Methods & Analysis</h2>
        <div class="explanation">
            <p><strong>Analyzing Data with PLINK</strong></p>
            <p>PLINK (pedigree data link) is a tool used to make sense of the genetic data obtained from DNA tests. Here's a simplified overview of how PLINK works:</p>
            <ul>
                <li>Preparing the Data: PLINK first gets the genetic data from the DNA test. It then gets the data ready for analysis by cleaning it up and organizing it.</li>
                <li>Checking the Data: Before doing any analysis, PLINK checks to make sure the data is good quality. It looks for any mistakes or problems and fixes them.</li>
                <li>Looking for Patterns: PLINK tries to find patterns in the genetic data. It looks at how different bits of DNA are related to each other and how they might be connected to things like traits or diseases.</li>
                <li>Comparing Groups: PLINK compares the genetic data of different groups of people. It looks for differences between these groups and tries to figure out what might be causing those differences.</li>
                <li>Visualizing the Results: Finally, PLINK helps visualize the results of the analysis. It creates charts and graphs that make it easier to understand what the data is telling us.</li>
            </ul>
        </div>

<div style="text-align:center;">
    <img src="https://carlalexander.ca/uploads/2018/09/sai-kiran-anagani-555972-unsplash.jpg" alt="High Performance Computing" style="width: auto; height: 300px; margin-right: 10px;">
    <img src="https://www.purestorage.com/content/dam/purestorage/knowledge/what-is-high-performance-computing.jpg.imgo.jpg" alt="High Performance Computing" style="width: auto; height: 300px; margin-right: 10px;">
</div>



<div class="explanation">
    <h3>Why PLINK Is Run on the Command-Line</h3>
    <p>PLINK is predominantly operated through the command-line for several reasons:</p>
    <ul>
        <li><strong>Efficiency:</strong> Command-line operations allow PLINK to process large datasets more efficiently than graphical user interfaces (GUIs). This is crucial in genomics, where datasets can be extremely large.</li>
        <li><strong>Reproducibility:</strong> Using command-line commands makes it easier to document and reproduce analyses, an essential aspect of scientific research. Scripts containing PLINK commands can be shared and executed by others, ensuring consistency in data analysis.</li>
        <li><strong>Flexibility:</strong> The command-line interface (CLI) enables users to easily combine PLINK with other tools and scripts for more comprehensive data analysis workflows.</li>
        <li><strong>Batch Processing:</strong> PLINK can be used in batch processing mode (processing multiple commands or analyses simultaneously), which is more seamlessly achieved via the command line.</li>
    </ul>
    <p>For these reasons, while PLINK can be daunting for beginners, its command-line interface is a powerful tool for performing complex genetic analyses efficiently and accurately.</p>
</div>

        
        <!-- Command-line operation block -->
        <div class="command-block">
            <p>#Example command for quality control of input data:</p>
            <code>plink --bfile yourdata --mind 0.1 --geno 0.1 --hwe 1e-6 --maf 0.05 --make-bed --out yourdata_qc </code>
            <p>#Example command for IBD estimation:</p>
            <code>plink --bfile yourdata_qc --genome --min 0.05 --out ibd_estimates</code>
        </div>



        <!-- Analysis Form -->
        <div class="explanation">
            <p><strong>Run Analysis</strong></p>
            <p>Click the button below to start the analysis. It may take a few seconds to complete.</p>
            <form id="analysisForm">
                <label for="inputData">Input Data:</label>
                <input type="file" id="inputData" accept=".txt" required>
                <BR>
                <BR>
                <button type="button" onclick="runAnalysis()">Run Analysis</button>
    <button type="button" onclick="runAnalysisWithDefaultData()">Run Analysis with Lazaridis et al. 2016 data</button>
            <p>The example data generated for this project was obtained form the following manuscript:<BR>Lazaridis, I., Nadel, D., Rollefson, G., Merrett, D. C., Rohland, N., Mallick, S., ... & Reich, D. (2016). Genomic insights into the origin of farming in the ancient Near East. Nature, 536(7617), 419-424.</p>
    
<!-- Your table, initially hidden -->
    <div id="analysisResults" style="display: none;">
        <h2>Analysis Results</h2>

<table border="1" style="width: 100%; table-layout: fixed;">
    <thead>
        <tr>
            <td>IID1</td>
            <td>IID2</td>
            <td>Z0</td>
            <td>Z1</td>
            <td>Z2</td>
            <td>PI_HAT</td>
            <td>DST</td>
            <td>PPC</td>
            <td>RATIO</td>
        </tr>
    </thead>
    <tr>
        <td>0-Frank Marchase</td>
        <td>1-Antonella Donato Marchase</td>
        <td>1</td>
        <td>0</td>
        <td>0</td>
        <td>0</td>
        <td>0.647035</td>
        <td>0</td>
        <td>1.4343</td>
    </tr>
    <tr>
        <td>0-Frank Marchase</td>
        <td>14-Vincent Callahan</td>
        <td>1</td>
        <td>0</td>
        <td>0</td>
        <td>0</td>
        <td>0.645127</td>
        <td>0</td>
        <td>1.4845</td>
    </tr>
    <tr>
        <td>0-Frank Marchase</td>
        <td>15-Louise Brennan Callahan</td>
        <td>1</td>
        <td>0</td>
        <td>0</td>
        <td>0</td>
        <td>0.645401</td>
        <td>0</td>
        <td>1.6762</td>
    </tr>
    <tr>
        <td>0-Frank Marchase</td>
        <td>27-Alice Sheehan Gazzola</td>
        <td>0.8414</td>
        <td>0.1586</td>
        <td>0</td>
        <td>0.0793</td>
        <td>0.665493</td>
        <td>1</td>
        <td>2.3643</td>
    </tr>
    <tr>
        <td>0-Frank Marchase</td>
        <td>28-Steven LaScala</td>
        <td>0.5968</td>
        <td>0.3227</td>
        <td>0.0806</td>
        <td>0.2419</td>
        <td>0.730862</td>
        <td>1</td>
        <td>3.8385</td>
    </tr>
    <tr>
        <td>0-Frank Marchase</td>
        <td>29-Karen Gazzola Marchase</td>
        <td>0.8244</td>
        <td>0.1756</td>
        <td>0</td>
        <td>0.0878</td>
        <td>0.672771</td>
        <td>1</td>
        <td>2.3992</td>
    </tr>
    <tr>
        <td>0-Frank Marchase</td>
        <td>30-Matt Marchase</td>
        <td>0.66</td>
        <td>0.2768</td>
        <td>0.0632</td>
        <td>0.2016</td>
        <td>0.719504</td>
        <td>1</td>
        <td>3.361</td>
    </tr>
    <tr>
        <td>1-Antonella Donato Marchase</td>
        <td>14-Vincent Callahan</td>
        <td>1</td>
        <td>0</td>
        <td>0</td>
        <td>0</td>
        <td>0.652104</td>
        <td>0</td>
        <td>1.2693</td>
    </tr>
    <tr>
        <td>1-Antonella Donato Marchase</td>
        <td>15-Louise Brennan Callahan</td>
        <td>1</td>
        <td>0</td>
        <td>0</td>
        <td>0</td>
        <td>0.632663</td>
        <td>0</td>
        <td>1.32</td>
    </tr>
    <tr>
        <td>1-Antonella Donato Marchase</td>
        <td>27-Alice Sheehan Gazzola</td>
        <td>1</td>
        <td>0</td>
        <td>0</td>
        <td>0</td>
        <td>0.652543</td>
        <td>0.0023</td>
        <td>1.8254</td>
    </tr>
    <tr>
        <td>1-Antonella Donato Marchase</td>
        <td>28-Steven LaScala</td>
        <td>0.638</td>
        <td>0.282</td>
        <td>0.0799</td>
        <td>0.221</td>
        <td>0.725737</td>
        <td>1</td>
        <td>3.176</td>
    </tr>
    <tr>
        <td>1-Antonella Donato Marchase</td>
        <td>29-Karen Gazzola Marchase</td>
        <td>0.8752</td>
        <td>0.1248</td>
        <td>0</td>
        <td>0.0624</td>
        <td>0.673939</td>
        <td>0.993</td>
        <td>2.1705</td>
    </tr>
    <tr>
        <td>1-Antonella Donato Marchase</td>
        <td>30-Matt Marchase</td>
        <td>0.7227</td>
        <td>0.2681</td>
        <td>0.0092</td>
        <td>0.1432</td>
        <td>0.700348</td>
        <td>1</td>
        <td>2.7208</td>
    </tr>
    <tr>
        <td>14-Vincent Callahan</td>
        <td>15-Louise Brennan Callahan</td>
        <td>1</td>
        <td>0</td>
        <td>0</td>
        <td>0</td>
        <td>0.639051</td>
        <td>0</td>
        <td>1.3809</td>
    </tr>
    <tr>
        <td>14-Vincent Callahan</td>
        <td>27-Alice Sheehan Gazzola</td>
        <td>0.4533</td>
        <td>0.5467</td>
        <td>0</td>
        <td>0.2734</td>
        <td>0.704415</td>
        <td>1</td>
        <td>4.3293</td>
    </tr>
    <tr>
        <td>14-Vincent Callahan</td>
        <td>28-Steven LaScala</td>
        <td>0.8626</td>
        <td>0.1374</td>
        <td>0</td>
        <td>0.0687</td>
        <td>0.667051</td>
        <td>0.9408</td>
        <td>2.1059</td>
    </tr>
    <tr>
        <td>14-Vincent Callahan</td>
        <td>29-Karen Gazzola Marchase</td>
        <td>0.6838</td>
        <td>0.3162</td>
        <td>0</td>
        <td>0.1581</td>
        <td>0.691463</td>
        <td>1</td>
        <td>2.8262</td>
    </tr>
    <tr>
        <td>14-Vincent Callahan</td>
        <td>30-Matt Marchase</td>
        <td>0.7849</td>
        <td>0.2151</td>
        <td>0</td>
        <td>0.1075</td>
        <td>0.676792</td>
        <td>1</td>
        <td>2.3832</td>
    </tr>
    <tr>
        <td>15-Louise Brennan Callahan</td>
        <td>27-Alice Sheehan Gazzola</td>
        <td>0.3989</td>
        <td>0.6011</td>
        <td>0</td>
        <td>0.3006</td>
        <td>0.722438</td>
        <td>1</td>
        <td>6.0067</td>
    </tr>
    <tr>
        <td>15-Louise Brennan Callahan</td>
        <td>28-Steven LaScala</td>
        <td>0.8543</td>
        <td>0.1457</td>
        <td>0</td>
        <td>0.0728</td>
        <td>0.666484</td>
        <td>1</td>
        <td>2.4976</td>
    </tr>
    <tr>
        <td>15-Louise Brennan Callahan</td>
        <td>29-Karen Gazzola Marchase</td>
        <td>0.6316</td>
        <td>0.3684</td>
        <td>0</td>
        <td>0.1842</td>
        <td>0.68786</td>
        <td>1</td>
        <td>3.3718</td>
    </tr>
    <tr>
        <td>15-Louise Brennan Callahan</td>
        <td>30-Matt Marchase</td>
        <td>0.6441</td>
        <td>0.3559</td>
        <td>0</td>
        <td>0.178</td>
        <td>0.688137</td>
        <td>1</td>
        <td>3.4525</td>
    </tr>
    <tr>
        <td>27-Alice Sheehan Gazzola</td>
        <td>28-Steven LaScala</td>
        <td>0.6189</td>
        <td>0.3811</td>
        <td>0</td>
        <td>0.1906</td>
        <td>0.69124</td>
        <td>1</td>
        <td>3.9762</td>
    </tr>
    <tr>
        <td>27-Alice Sheehan Gazzola</td>
        <td>29-Karen Gazzola Marchase</td>
        <td>0</td>
        <td>1</td>
        <td>0</td>
        <td>0.5</td>
        <td>0.739465</td>
        <td>1</td>
        <td>21.631</td>
    </tr>
    <tr>
        <td>27-Alice Sheehan Gazzola</td>
        <td>30-Matt Marchase</td>
        <td>0.2618</td>
        <td>0.7382</td>
        <td>0</td>
        <td>0.3691</td>
        <td>0.728384</td>
        <td>1</td>
        <td>9.8116</td>
    </tr>
    <tr>
        <td>28-Steven LaScala</td>
        <td>29-Karen Gazzola Marchase</td>
        <td>0.5675</td>
        <td>0.4325</td>
        <td>0</td>
        <td>0.2162</td>
        <td>0.696591</td>
        <td>1</td>
        <td>4.3273</td>
    </tr>
    <tr>
        <td>28-Steven LaScala</td>
        <td>30-Matt Marchase</td>
        <td>0.0811</td>
        <td>0.6707</td>
        <td>0.2482</td>
        <td>0.5835</td>
        <td>0.829061</td>
        <td>1</td>
        <td>42.4356</td>
    </tr>
    <tr>
        <td>29-Karen Gazzola Marchase</td>
        <td>30-Matt Marchase</td>
        <td>0</td>
        <td>1</td>
        <td>0</td>
        <td>0.5</td>
        <td>0.752385</td>
        <td>1</td>
        <td>30.3185</td>
    </tr>
</table>

            </form>
        </div>

        <!-- Loading bar -->
        <div class="loading-bar">
            <div class="progress-bar"></div>
        </div>
    </div>


<div id="downloadButton" style="display: none;">
    <button onclick="exportTableToCSV('relatedness_analysis.csv')">Download as CSV</button>
</div>



</div>
    <!-- PLINK Output Explained Section -->
<div id="plink" class="tab">
    <h2>Understanding PLINK Output in Genetic Relatedness Analysis</h2>
    <div class="explanation" style="color: black;">
        <p><strong>Identity-by-Descent (IBD) Estimation:</strong> In ancestry and relatedness studies, PLINK estimates the proportion of the genome that individuals share. The key metrics include:</p>
        <ul>
            <li><strong>PI_HAT:</strong> The estimated proportion of the genome shared by two individuals. Higher values indicate closer genetic relatedness.</li>
            <li><strong>Z0, Z1, Z2:</strong> These represent the estimated proportions of the genome where two individuals share 0, 1, or 2 alleles IBD (identity by descent), respectively.</li>
        </ul>
        <p><strong>Column Headers in PLINK Output:</strong></p>
        <ul>
            <li><strong>FID1, IID1:</strong> Family and individual ID for the first individual.</li>
            <li><strong>FID2, IID2:</strong> Family and individual ID for the second individual.</li>
            <li><strong>RT:</strong> Relationship type. OT means other, or relationship unknown. </li>
            <li><strong>EZ:</strong> Empirical z-score.This represents a statistical measure of the observed genetic similarity between two individuals compared to what would be expected by chance. It is computed based on the allele sharing observed between pairs of individuals and is used to assess the significance of genetic relatedness or similarity.</li>
            <li><strong>Z0, Z1, Z2:</strong> Estimated proportions of genome sharing 0, 1, or 2 alleles IBD.</li>
            <li><strong>PI_HAT:</strong> Estimated proportion of the genome shared by the two individuals.</li>
            <li><strong>PHE:</strong> Phenotype (if available).</li>
            <li><strong>DST:</strong> Genetic distance between the two individuals.</li>
            <li><strong>PPC:</strong> Posterior probability of the relationship being correctly specified.</li>
            <li><strong>RATIO:</strong> Ratio of the likelihood of the observed data given the relationships being specified versus relationships not being specified.</li>
        </ul>
        <p><strong>Applications:</strong> These metrics help in constructing family trees, identifying population structures, estimating individual ancestries, and assessing relatedness between individuals in genetic studies.</p>

        <!-- Add more content as needed -->
    

<table border="1" cellspacing="0" cellpadding="5">
    <thead>
        <tr>
            <th>Relationship</th>
            <th>Proportion Matching DNA (IBD) (PI_HAT)</th>
            <th>Proportion IBD1 (Z1)</th>
            <th>Proportion IBD1 (Z2)</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Self/Twin</td>
            <td>1.0</td>
            <td>0.0</td>
            <td>1.0</td>
        </tr>
        <tr>
            <td>Parent/Child</td>
            <td>0.5 (0.4 - 0.6)</td>
            <td>1.0</td>
            <td>0.0</td>
        </tr>
        <tr>
            <td>Full Sibling</td>
            <td>0.5 (0.4 - 0.6)</td>
            <td>0.5</td>
            <td>0.25</td>
        </tr>
        <tr>
            <td>Grandparent/Grandchild<BR>Aunt/Uncle<BR>Half-Sibling</td>
            <td>0.25 (0.15 - 0.4)</td>
            <td>0.5</td>
            <td>0.0</td>
        </tr>
        <tr>
            <td>1st Cousin</td>
            <td>0.125 (0.1 - 0.15)</td>
            <td>0.25</td>
            <td>0.0</td>
        </tr>

    </tbody>
</table>
<a href="https://isogg.org/wiki/Autosomal_DNA_statistics">https://isogg.org/wiki/Autosomal_DNA_statistics</a>
<BR>
<a href="https://isogg.org/wiki/Autosomal_DNA_statistics#Table">https://isogg.org/wiki/Autosomal_DNA_statistics#Table</a>
<BR>
<a href="https://thegeneticgenealogist.com/wp-content/uploads/2020/03/Shared-cM-Project-Relationship-Chart.png">https://thegeneticgenealogist.com/wp-content/uploads/2020/03/Shared-cM-Project-Relationship-Chart.png</a>
</div>
</div>




    <!-- JavaScript to handle tab switching and pretend analysis -->
    <script>
function openTab(tabName) {
    // Hide all tab content elements
    var tabs = document.getElementsByClassName("tab");
    for (var i = 0; i < tabs.length; i++) {
        tabs[i].style.display = "none";
    }

    // Show the selected tab content
    document.getElementById(tabName).style.display = "block";
}




        // Function to pretend running the analysis and show progress
        function runAnalysis() {
            var form = document.getElementById('analysisForm');
            var fileInput = document.getElementById('inputData');
            if (fileInput.files.length > 0) {
                var loadingBar = document.querySelector('.loading-bar');
                var progressBar = document.querySelector('.progress-bar');
                loadingBar.style.display = 'block';
                progressBar.style.width = '0%';
                var progress = 0;
                var interval = setInterval(function() {
                    progress += 10;
                    progressBar.style.width = progress + '%';
                    if (progress >= 100) {
                        clearInterval(interval);
                        alert('Analysis Complete!');
                        loadingBar.style.display = 'none';

document.getElementById('analysisResults').style.display = 'block';
document.getElementById('downloadButton').style.display = 'block';
                    }
                }, 200); // Adjust time here for longer or shorter simulation
            } else {
                alert('Please select a data file to analyze.');
            }
        }

        // Initialize the first tab open
        document.addEventListener('DOMContentLoaded', function() {
            openTab('background');
        });


function runAnalysisWithDefaultData() {
    var loadingBar = document.querySelector('.loading-bar');
    var progressBar = document.querySelector('.progress-bar');
    loadingBar.style.display = 'block';
    progressBar.style.width = '0%';
    var progress = 0;
    var interval = setInterval(function() {
        progress += 10;
        progressBar.style.width = progress + '%';
        if (progress >= 100) {
            clearInterval(interval);
            alert('Analysis Complete with Default Data!');
            loadingBar.style.display = 'none';
            document.getElementById('analysisResults').style.display = 'block';
            document.getElementById('downloadButton').style.display = 'block';
        }
    }, 200); // Adjust time as needed for the simulated loading effect
}



function exportTableToCSV(filename) {
    var csv = [];
    var rows = document.querySelectorAll("#analysisResults table tr");
    
    for (var i = 0; i < rows.length; i++) {
        var row = [], cols = rows[i].querySelectorAll("td, th");
        
        for (var j = 0; j < cols.length; j++) 
            row.push('"' + cols[j].innerText + '"');
        
        csv.push(row.join(","));        
    }

    // Download CSV
    downloadCSV(csv.join("\n"), filename);
}

function downloadCSV(csv, filename) {
    var csvFile;
    var downloadLink;

    // CSV file
    csvFile = new Blob([csv], {type: "text/csv"});

    // Download link
    downloadLink = document.createElement("a");

    // File name
    downloadLink.download = filename;

    // Create a link to the file
    downloadLink.href = window.URL.createObjectURL(csvFile);

    // Hide download link
    downloadLink.style.display = "none";

    // Add the link to DOM
    document.body.appendChild(downloadLink);

    // Click download link
    downloadLink.click();
}


    </script>
</body>
</html>

